### DISCUSSION TITLE
[Help]: Question about CSS rules used

### DISCUSSION THREAD

**Discussion by meirroth** [davidjerleke/embla-carousel#838] [Help] [answered]
### Summary

Hi @davidjerleke I'm about to deploy Embla slides to prod 😃 just was curios to know the reason behind these two CSS rules, maybe you remember?
- [backface-visibility: hidden; on embla__container](https://github.com/davidjerleke/embla-carousel/commit/85a3d6d8974e93f2d5678b82a5d37b2e9ee662cb#diff-44dc5e14d8310eadca9db940aa4ae252e016bbb38d8959ff526fc7747843af37R36)
- [min-width: 0; on embla__slide](https://github.com/davidjerleke/embla-carousel/commit/45684813ddc4eecf4eb0e91a666dc055343471fe#diff-67b917e276423be393c8a561579eb5ef614167dcf73b21659f07ece710e9ce59R43)

I think I figured out why touch-action: pan-y; on embla__container.

Thank you,
Meir

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [ ] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by davidjerleke** [comment, marked as answer]
Hi @meirroth,

Thanks for your question. 

> [backface-visibility: hidden; on embla__container](https://github.com/davidjerleke/embla-carousel/commit/85a3d6d8974e93f2d5678b82a5d37b2e9ee662cb#diff-44dc5e14d8310eadca9db940aa4ae252e016bbb38d8959ff526fc7747843af37R36)

`backface-visibility: hidden;` is there because sometimes some browsers struggle with rendering transforms, which Embla is using to create its scroll effect. For example it can fix unwanted flickering and/or jagged edges. Please read this [StackOverflow answer](https://stackoverflow.com/a/28512658/4647781 ) to understand it better.

> [min-width: 0; on embla__slide](https://github.com/davidjerleke/embla-carousel/commit/45684813ddc4eecf4eb0e91a666dc055343471fe#diff-67b917e276423be393c8a561579eb5ef614167dcf73b21659f07ece710e9ce59R43)

This is here to prevent slides and slide content to grow larger than their slide container. Please read about why this is necessary [here](https://www.bigbinary.com/blog/understanding-the-automatic-minimum-size-of-flex-items).

> I think I figured out why touch-action: pan-y; on embla__container.

`touch-action: pan-y;` should be applied to horizontal carousels to prevent the page from scrolling vertically while scrolling a horizontal carousel, which is horrible UX 🙂. Note that a vertical carousel should have `touch-action: pan-x;` instead. This property is relevant for touch devices.

Best,
David

**Reply by meirroth** [reply]
@davidjerleke Thank you for clarifying!
